src/DataStructures/TempBuffer.hpp

```diff
@@ -39,3 +39,16 @@ template <typename TagList>
 struct TempBuffer<TagList, false> : Variables<TagList> {
   using Variables<TagList>::Variables;
 };
+
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief A common operation for checking the size and resizing if neccessary
+ * for a buffer or return passed by not_null
```

Comment:
1. How about the more concise: "Resize a memory buffer if needed to ensure that it has the desired size" ?
2. I like the idea of this utility... but we should make sure to only use this when the UNLIKELY really is UNLIKELY, or we risk hurting performance. Perhaps add a doxygen note to this effect?